contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,334 @@
+        address indexed player,
+        uint256 stake,
+        uint256 amount
+    );
+
+    event PlayerTwoMoves(
+        bytes32 indexed gameKey,
+        address indexed player,
+        uint256 amount,
+        bytes32 moveHash,
+        uint256 expiryDate
+    );
+
+    event PlayerOneReveals(
+        bytes32 indexed gameKey,
+        address indexed player,
+        Moves move,
+        uint256 expiryDate
+    );
+
+    event PlayerTwoReveals(
+        bytes32 indexed gameKey,
+        address indexed player,
+        Moves move
+    );
+
+    event DrawRefund(
+        bytes32 indexed gameKey,
+        address indexed playerOne,
+        address indexed playerTwo,
+        uint256 playerOneRefund,
+        uint256 playerTwoRefund
+    );
+
+    event GamePayment(
+        bytes32 indexed gameKey,
+        address indexed player,
+        uint256 amount
+    );
+
+    event ForfeitPaid(
+        bytes32 indexed gameKey,
+        address indexed player,
+        uint256 amount
+    );
+
+    event WithDraw(
+        address indexed player,
+        uint amount
+    );
+
+    event PlayerOneEndsGame(
+        bytes32 indexed gameKey,
+        address indexed player,
+        uint256 amount
+    );
+
+    // The creator's move is embedded in the game key
+    function movePlayerOne(bytes32 gameKey, uint256 stake) external payable whenNotPaused {
+
+        require(gameKey != bytes32(0), "Game key hash is empty");
+        Game storage game = games[gameKey];
+        require(game.step == Steps.UNSTARTED, INVALID_STEP_MSG);
+        require(game.playerOne == NULL_ADDRESS, INVALID_MOVE_MSG);
+
+        game.stake = stake;
+        game.playerOne = msg.sender;
+        game.step = Steps.PLAYER_ONE_MOVED;
+        depositStake(stake);
+
+        emit PlayerOneMoves(gameKey, msg.sender, stake, msg.value);
+    }
+
+    function movePlayerTwo(bytes32 gameKey, bytes32 moveHash) external payable whenNotPaused {
+
+        require(moveHash != bytes32(0), "Move hash is empty");
+        Game storage game = games[gameKey];
+        require(game.step == Steps.PLAYER_ONE_MOVED, INVALID_STEP_MSG);
+
+        uint256 expiryDate = block.timestamp.add(FORFEIT_WINDOW);
+        game.playerTwo = msg.sender;
+        game.playerTwoMoveHash = moveHash;
+        game.step = Steps.PLAYER_TWO_MOVE;
+        game.expiryDate = expiryDate;
+        depositStake(game.stake);
+
+        emit PlayerTwoMoves(gameKey, msg.sender, msg.value, moveHash, expiryDate);
+    }
+
+    function depositStake(uint256 stake) internal {
+
+        if (msg.value == stake) {
+            return;
+        }
+
+        uint256 senderBalance = balances[msg.sender];
+        if (msg.value < stake) {
+            require(senderBalance >= stake.sub(msg.value), "Insufficient balance");
+        }
+
+        balances[msg.sender] = senderBalance.add(msg.value).sub(stake);
+    }
+
+    function revealPlayerOne(bytes32 secret, uint8 playerOneMove) external whenNotPaused {
```

Comment:
Have you tried declaring `, Moves playerOneMove)`?

Reply:
Sorry. I somehow overlooked this. As a consequence, I changed the following:

`require(Moves(playerOneMove) >= Moves.ROCK && Moves(playerOneMove) <= Moves.SCISSORS, INVALID_MOVE_MSG);`

To

`require(Moves(playerOneMove) >= Moves.ROCK, INVALID_MOVE_MSG);`

If I send 4, then `Moves playerOneMove` enum type hint catches the fact the value is out of bounds making the second conditional redundant.

Reply:
I think, you can winnow it down to:

```solidity
require(playerOneMove >= Moves.ROCK, INVALID_MOVE_MSG);
```

Reply:
Yes. I fixed.